schema_based_on_diagram.sql

```diff
@@ -0,0 +1,51 @@
+CREATE TABLE "patients"(
+    "id" INTEGER NOT NULL,
+    "name" VARCHAR(255) NOT NULL,
+    "date_of_birth" DATE NOT NULL
+);
+ALTER TABLE
+    "patients" ADD PRIMARY KEY("id");
+CREATE TABLE "invoices"(
+    "id" INTEGER NOT NULL,
+    "total_amount" DECIMAL(8, 2) NOT NULL,
+    "generated_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "payed_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "medical_history_id" INTEGER NOT NULL
+);
+ALTER TABLE
+    "invoices" ADD PRIMARY KEY("id");
+CREATE TABLE "medical_histories"(
+    "id" INTEGER NOT NULL,
+    "admitted_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "patient_id" INTEGER NOT NULL,
+    "status" VARCHAR(255) NOT NULL
+);
+ALTER TABLE
+    "medical_histories" ADD PRIMARY KEY("id");
+CREATE TABLE "treatments"(
+    "id" INTEGER NOT NULL,
+    "type" VARCHAR(255) NOT NULL,
+    "name" VARCHAR(255) NOT NULL
+);
+ALTER TABLE
+    "treatments" ADD PRIMARY KEY("id");
+CREATE TABLE "invoice_items"(
+    "id" INTEGER NOT NULL,
+    "unit_price" DECIMAL(8, 2) NOT NULL,
+    "quantity" INTEGER NOT NULL,
+    "total_price" DECIMAL(8, 2) NOT NULL,
+    "invoice_id" INTEGER NOT NULL,
+    "treatment_id" INTEGER NOT NULL
+);
+ALTER TABLE
+    "invoice_items" ADD PRIMARY KEY("id");
+ALTER TABLE
+    "medical_histories" ADD CONSTRAINT "medical_histories_patient_id_foreign" FOREIGN KEY("patient_id") REFERENCES "patients"("id");
```

Comment:
- You have done great so far :+1: Following the reviewers' suggestion and creating the tables from the diagram. :+1: Before moving to the next milestone, there are still some requirements, you need to meet. :+1: :100: 

> Join tables from many-to-many relationships might not appear in the diagram, but you still need them.

- [ ] As stated in the project requirement, you need to implement many-to-many relationships using join table even if the table is not visible diagram. Looking at the diagram. you can see there is a `many-to-many relationship` between `medical_histories` and `treatments` table. On that note, you need to create a join table to represent that relationship. here's a suggested code snippet: :+1: :smile: :+1: 

```sh
CREATE TABLE medical_history_treatments (
  medical_history_id INT,
  treatment_id INT,
  PRIMARY KEY (medical_history_id, treatment_id),
  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id),
  FOREIGN KEY (treatment_id) REFERENCES treatments (id)
);
```



> Remember to add the FK indexes.

- [ ] Also as stated in the project requirement, you need to create foreign for all the foreng keys in each table. currently, you don't have that implemented. I'll suggest you implnent that. :+1: :smile:  Here's a code snippet to guide you through:
```sh
CREATE INDEX idx_medhist_id ON medical_history_treatments (medical_history_id);

CREATE INDEX idx_treatment_id ON medical_history_treatments (treatment_id);

CREATE INDEX idx_invoice_items_invoice_id ON invoice_items (invoice_id);

CREATE INDEX idx_invoice_items_treatment_id ON invoice_items (treatment_id);

CREATE INDEX idx_invoices_medicalhistory_id ON invoices (medical_history_id);

CREATE INDEX idx_patient_id ON medical_histories (patient_id);
```